src/Bootstrappers/Integrations/PostgresTenancyBootstrapper.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Bootstrappers\Integrations;
+
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Database\DatabaseManager;
+use Stancl\Tenancy\Contracts\TenancyBootstrapper;
+use Stancl\Tenancy\Contracts\Tenant;
+use Stancl\Tenancy\Database\Contracts\TenantWithDatabase;
+
+class PostgresTenancyBootstrapper implements TenancyBootstrapper
+{
+    protected array $originalCentralConnectionConfig;
+    protected array $originalPostgresConfig;
+
+    public function __construct(
+        protected Repository $config,
+        protected DatabaseManager $database,
+    ) {
+        $this->originalCentralConnectionConfig = $config->get('database.connections.central');
+        $this->originalPostgresConfig = $config->get('database.connections.pgsql');
+    }
+
+    public function bootstrap(Tenant $tenant): void
+    {
+        /** @var TenantWithDatabase $tenant */
+        $this->database->purge('central');
+
+        $this->config->set('database.connections.pgsql.username', $tenant->getTenantKey());
```

Comment:
Why `getTenantKey()` instead of `$tenant->database()->getUsername()`?

Reply:
These things should be in line with the existing logic we have (eg how we do permission controlled DB managers).

Reply:
I see, `$tenant->database()->getUsername()` makes more sense. Changed`$tenant->getTenantKey()` to `$tenant->database()->getUsername() ?? $tenant->getTenantKey()`. So we only default to the tenant key if the db username is null